### ISSUE TITLE
Add better offset configuration options (e.g. by display)

### ISSUE THREAD

**Issue by kinglouie** [koekeishiya/chunkwm#430]
At the current state it is only possible to configure offsets globally or by space number.
This is a little annoying when using a macbook with/without an external montitor. My aim is to have big offsets and gaps on my external 4k monitor and low/no offsets/gaps on the builtin macbook display. At the moment I solve this by having a fixed amount of spaces on the builtin display and do the configuration by space number, however when disconnecting the external monitor this falls apart because now all spaces are on the builtin display. The current workaround is to toggle the offsets to zero but then my Übersicht bar is hidden  behind the windows.

It would be nice if there was some way to define a callback function which acts on different environmental params like this:
:
```
function eval_rules(num_displays, display_id, space_id) {
    if(num_displays > 1 && display_id == 2)  {
        chunkc set desktop_offset_top     100
        chunkc set desktop_offset_bottom  80
        chunkc set desktop_offset_left    190
        chunkc set desktop_offset_right   190
        chunkc set desktop_offset_gap     60
    } else {
        chunkc set desktop_offset_top     23  // for übersicht bar
        chunkc set desktop_offset_bottom  0
        chunkc set desktop_offset_left    0
        chunkc set desktop_offset_right   0
        chunkc set desktop_offset_gap     0
    }
}
```

and call this callback function for every space and on every change of the provided environment variables.
This would allow a lot more sophisticated configurations that are also more expressive then configuring hardcoded space ids.

**Comment by chrishoage** [comment]
`chunkwmrc` is a bash script. In theory, this script could be run on monitor addition and deletion (and possibly on space creation / deletion)  

In `chunkwmrc` we could use the `query` system to detect the current state of the system and be used in the rc file to set up the configuration appropriately

IIRC this is how bspwmrc works

**Comment by kinglouie** [comment]
You're right, I just checked bspwms examples and found [this example](https://github.com/baskerville/bspwm/tree/master/examples/external_rules) how this configuration feature could be designed 👍
